src/app/report/report-value/report-value.component.ts

```diff
@@ -0,0 +1,445 @@
+        this.getRealEditedValueForNullable(this.editedTransformation),
+        'Transformation',
+        true,
+      )
+      .nonNullableVariable(this.report!.stubStrategy, this.editedReportStubStrategy, 'Report level stub strategy');
+    const editedVariables = this.getRealEditedVariables();
+    const originalVariableNames: string[] = this.originalVariables.map((v) => v.name);
+    const editedVariableNames: string[] = editedVariables.map((v) => v.name);
+    result.nonNullableVariable(
+      originalVariableNames.join('\n'),
+      editedVariableNames.join('\n'),
+      'Defined variables (ordered)',
+      true,
+    );
+    for (const n of ReportValueComponent.getCommonVariableNames(originalVariableNames, editedVariableNames)) {
+      const originalValue = this.originalVariables.find((v) => v.name === n)!.value;
+      const editedValue = editedVariables.find((v) => v.name === n)!.value;
+      result.nonNullableVariable(originalValue, editedValue, `Variable ${n}`, true);
+    }
+    return result;
+  }
+
+  // TODO: Include in Karma tests.
+  requestSave(): void {
+    if (!this.isEdited()) {
+      throw new Error('ReportValueComponent.save() should not be called when the report node has not been edited');
+    }
+    const updateReport: UpdateReport = {};
+    if (this.editedName !== this.report!.name) {
+      updateReport.name = this.editedName;
+    }
+    const description: string | null = this.getRealEditedValueForNullable(this.editedDescription);
+    if (description !== this.report!.description) {
+      updateReport.description = description;
+    }
+    const path: string | null = this.getRealEditedValueForNullable(this.editedPath);
+    if (path !== this.report!.path) {
+      updateReport.path = path;
+    }
+    const transformation = this.getRealEditedValueForNullable(this.editedTransformation);
+    if (transformation !== this.report!.transformation) {
+      updateReport.transformation = transformation;
+    }
+    if (!this.hasNoUnsavedVariables()) {
+      updateReport.variables = this.getVariablesUpdate();
+    }
+    if (this.editedReportStubStrategy !== this.report!.stubStrategy) {
+      updateReport.stubStrategy = this.editedReportStubStrategy;
+    }
+    this.save.emit({ updateReport });
+  }
+
+  // For testing purposes
+  setVariables(variables: Variable[]): void {
+    this.originalVariables = variables;
+    this.editedVariables = ReportValueComponent.calculateEditedVariables(this.originalVariables);
+    this.refreshDuplicateVariables();
+  }
+
+  protected onDownload(downloadOptions: DownloadOptions): void {
+    this.downloadRequest.emit(downloadOptions);
+  }
+
+  private newReport(report: PartialReport): void {
+    // When a report is edited, the update report is received from a
+    // subscription. Reacting on a subscription is not automatically
+    // in the Angular zone, so we have to run in the Angular zone
+    // explicitly.
+    this.ngZone.run(() => {
```

Comment:
Er zijn manieren om wel een correct updatende template te hebben zonder ngZone te moeten aanroepen. objects vervangen alsof ze immutable zijn niet direct de waarde van observables in de template gebruiken bijvoorbeeld.

ngZone checkt variable references of ze geupdate zijn, als je meer wilt weten kunnen we een keer samen zitten hierover